overlord/snapstate/refresh_control.go

```diff
@@ -0,0 +1,173 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
```

Comment:
we need to think how to call this feature internally.  refresh-control referred to the old validation already

Reply:
maybe it should be called "autorefresh_gating.go" ?